packages/office-ui-fabric-react/src/components/Shimmer/examples/Shimmer.Application.Example.tsx

```diff
@@ -31,140 +41,92 @@ const fileIcons: { name: string }[] = [
   { name: 'xsn' },
 ];
 
-const ITEMS_COUNT = 200;
-const INTERVAL_DELAY = 2500;
-
-let _items: IExampleItem[];
-
-export interface IShimmerApplicationExampleState {
-  items: IExampleItem[]; // DetailsList `items` prop is required so it expects at least an empty array.
-  columns?: IColumn[];
-  isDataLoaded?: boolean;
-}
-
-export class ShimmerApplicationExample extends React.Component<{}, IShimmerApplicationExampleState> {
-  private _lastIntervalId: number;
-  private _lastIndexWithData: number;
-  private _async: Async;
-
-  constructor(props: {}) {
-    super(props);
-
-    this.state = {
-      items: [],
-      columns: _buildColumns(),
-      isDataLoaded: false,
-    };
+const randomFileIcon = (): { docType: string; url: string } => {
+  const docType: string = fileIcons[Math.floor(Math.random() * fileIcons.length) + 0].name;
+  return {
+    docType,
+    url: `https://static2.sharepointonline.com/files/fabric/assets/item-types/16/${docType}.svg`,
+  };
+};
 
-    this._async = new Async(this);
+const buildShimmerColumns = (): IColumn[] => {
+  const currentItems = createListItems(1);
+  const columns: IColumn[] = buildColumns(currentItems);
+  for (const column of columns) {
+    if (column.key === 'thumbnail') {
+      column.name = 'FileType';
+      column.minWidth = 16;
+      column.maxWidth = 16;
+      column.isIconOnly = true;
+      column.iconName = 'Page';
+      break;
+    }
   }
+  return columns;
+};
 
-  public componentWillUnmount(): void {
-    this._async.dispose();
-  }
+const shimmerColumns: IColumn[] = buildShimmerColumns();
```

Comment:
This and the items should be done within the component using `useConst(() => /*initialize*/)`